app/src/events/watcher.py

```
@@ -0,0 +1,111 @@
+        self.callback = callback
+        self.debounce_seconds = debounce_seconds
+        self._observer: Optional[Observer] = None
+        self._running = False
+
+    def start(self):
+        """Start watching for file changes."""
+        if self._running:
+            return
+
+        if not self.watch_path.exists():
+            raise FileNotFoundError(f"Watch path does not exist: {self.watch_path}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)